packages/api/src/cli/codegen/languages/typescript.ts

```diff
@@ -189,80 +163,114 @@ export default class TSGenerator extends CodeGeneratorLanguage {
    *
    */
   async generator() {
-    const { operations, methods } = await this.loadOperationsAndMethods();
+    const sdkSource = this.createSourceFile();
 
-    const sdkSource = this.project.createSourceFile('index.ts', '');
+    if (Object.keys(this.schemas).length) {
+      this.createSchemasFile();
+      this.createTypesFile();
 
-    sdkSource.addImportDeclarations([
+      // Export all of our available types so they can be used in SDK implementations.
+      //
+      // We're exporting all of the types individually because TS has no way right now of allowing
+      // us to do `export type * from './types'` on a non-named entry.
+      //
+      // https://github.com/microsoft/TypeScript/issues/37238
+      const types = Array.from(this.types.keys());
+      types.sort();
+
+      sdkSource.addExportDeclarations([
+        {
+          isTypeOnly: true,
+          namedExports: types,
+          moduleSpecifier: './types',
+        },
+      ]);
+    } else {
+      // If we don't have any schemas then we shouldn't import a `types` file that doesn't exist.
+      sdkSource
+        .getImportDeclarations()
+        .find(id => id.getText() === "import type * as types from './types';")
```

Comment:
Because we create the source file first, which does operation and response processing which in turn creates our schemas, we're inserting this import into the source file without having knowledge if we've actually got schemas for the spec we're using. This here is just cleaning that up so we aren't trying to import a file that we never actually made.